tools/clang/test/CodeGenSPIRV/vk.layout.cbuffer.fxc.matrix.struct.hlsl

```diff
@@ -0,0 +1,58 @@
+// Run: %dxc -T ps_6_0 -E main -fvk-use-dx-layout
+
+// CHECK: OpDecorate [[arr_f2:%\w+]] ArrayStride 16
+// CHECK: OpMemberDecorate %layout 0 Offset 0
+// CHECK: OpMemberDecorate %layout 1 Offset 16
+// CHECK: OpMemberDecorate %layout 2 Offset 36
+// CHECK: OpMemberDecorate %type_buffer0 0 Offset 0
+// CHECK: OpMemberDecorate %type_buffer0 1 Offset 16
+// CHECK: OpMemberDecorate %type_buffer0 2 Offset 56
+
+// CHECK: [[arr_f2]] = OpTypeArray %float %uint_2
+// CHECK: %layout = OpTypeStruct %float [[arr_f2]] %float
+// CHECK: %type_buffer0 = OpTypeStruct %float %layout %float
+// CHECK: %_ptr_Uniform_type_buffer0 = OpTypePointer Uniform %type_buffer0
+
+// CHECK: [[layout_clone:%\w+]] = OpTypeStruct %float %v2float %float
+// CHECK: [[type_buffer0_clone:%\w+]] = OpTypeStruct %float [[layout_clone]] %float
+// CHECK: [[ptr_type_buffer0_clone:%\w+]] = OpTypePointer Private [[type_buffer0_clone]]
+
+// CHECK: %buffer0 = OpVariable %_ptr_Uniform_type_buffer0 Uniform
+// CHECK: [[buffer0_clone:%\w+]] = OpVariable [[ptr_type_buffer0_clone]] Private
+
+cbuffer buffer0 {
+  float dummy0;                      // Offset:    0 Size:     4 [unused]
+  struct layout
+  {
+      float1x1 dummy0;               // Offset:   16
+      float1x2 foo;                  // Offset:   32
+      float end;                     // Offset:   52
+
+  } bar;                             // Offset:   16 Size:    40 [unused]
+  float end;                         // Offset:   56 Size:     4
+};
+
+// CHECK: %module_init = OpFunction %void
```

Comment:
Where is this `module.init` function called from? Can you add the CHECK for `OpFunctionCall` to it?